framework/math/functions/response_function.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include "framework/math/functions/function.h"
+#include "framework/mesh/mesh_vector.h"
+
+namespace opensn
+{
+
+/**
+ * Base class for evaluating response functions given spatial location and material ID
+ */
+class ResponseFunction : public Function
+{
+public:
+  static InputParameters GetInputParameters();
+  explicit ResponseFunction(const InputParameters& params);
+
+  /**
+   * Evaluate this function
+   *
+   * \param num_groups Number of groups
+   * \param pos The xyz coordinates of the point where the function is evaluated.
+   * \param mat_id Material ID
+   * \return Vector with the response (should have `num_groups` entries)
+   */
+  virtual std::vector<double> Evaluate(int num_groups, const Vector3& pos, int mat_id) const = 0;
```

Comment:
This is a trivial detail, but could we change pos to xyz for consistency?

Reply:
Yes, that's no problem.